--- /home/phl/microchip/usb_device.c	2014-01-22 10:56:19.444278747 -0700
+++ usb_device.c	2013-12-30 13:35:28.000000000 -0700
@@ -254,7 +254,7 @@ firmware design flexibility.
 #include "./USB/usb.h"
 #include "HardwareProfile.h"
 
-#include "../USB/usb_device_local.h"
+#include "./USB/usb_device_local.h"
 
 
 #if defined(USB_USE_MSD)
@@ -363,7 +363,7 @@ volatile BYTE CtrlTrfData[USB_EP0_BUFF_S
     USB_USER_CONFIG_DESCRIPTOR_INCLUDE;
 #endif
 
-extern ROM BYTE *ROM USB_SD_Ptr[];
+extern const BYTE const **USB_SD_Ptr;
 
 /** DECLARATIONS ***************************************************/
 #if defined(__18CXX)
@@ -2213,11 +2213,11 @@ static void USBStdGetDscHandler(void)
         {
             case USB_DESCRIPTOR_DEVICE:
                 #if !defined(USB_USER_DEVICE_DESCRIPTOR)
-                    inPipes[0].pSrc.bRom = (ROM BYTE*)&device_dsc;
+                    inPipes[0].pSrc.bRom = (ROM BYTE*)pnt_device_dsc;
                 #else
                     inPipes[0].pSrc.bRom = (ROM BYTE*)USB_USER_DEVICE_DESCRIPTOR;
                 #endif
-                inPipes[0].wCount.Val = sizeof(device_dsc);
+                inPipes[0].wCount.Val = sizeof(USB_DEVICE_DESCRIPTOR);
                 break;
             case USB_DESCRIPTOR_CONFIGURATION:
                 #if !defined(USB_USER_CONFIG_DESCRIPTOR)
